### ISSUE TITLE
Marks as circular non-circular

### ISSUE THREAD

**Issue by z1ne2wo** [sindresorhus/safe-stringify#2]
Minimal example:
```
const o = { v: 1 };
console.log(safeStringify({ a: o, b: o })); // {"a":{"v":1},"b":"[Circular]"}
console.log(JSON.stringify({ a: o, b: o })); // {"a":{"v":1},"b":{"v":1}}
```

**Comment by masontikhonov** [comment]
Hi @sindresorhus !
I've found this repo while reading through the sindresorhus.com, and it hooked me with its simplicity.
When I started reading the code, I couldn't understand the purpose of  `seen.get(value) !== key` check, so I dived deeper into the commits history and found this issue.

But I still can't get the meaning of such a check (`seen.get(value) !== key`). Now I would say it allows us to avoid _some of_ false-positives:
```JavaScript
const o = { v: 1 };
console.log(safeStringify({ a: o, b: { a: o } })); // {"a":{"v":1},"b":{"a":{"v":1}}} — now it works!
console.log(JSON.stringify({ a: o, b: { a: o } })); // {"a":{"v":1},"b":{"a":{"v":1}}}
```

But the example mentioned by @z1ne2wo is still false-positive:

```JavaScript
const o = { v: 1 };
console.log(safeStringify({ a: o, b: o })); // {"a":{"v":1},"b":"[Circular]"}
console.log(JSON.stringify({ a: o, b: o })); // {"a":{"v":1},"b":{"v":1}}
```

So changing WeakSet to WeakMap with additional key check was only done to address this issue, or is there another purpose I can't figure out?

---
Sorry if my question seems dumb, I'm just a curious beginner.
